framework/auth/views.py

```diff
@@ -108,14 +108,16 @@ def forgot_password_get(auth, *args, **kwargs):
 # Log in
 ###############################################################################
 
-# TODO: Rewrite async
 @collect_auth
 def auth_login(auth, **kwargs):
     """If GET request, show login page. If POST, attempt to log user in if
     login form passsed; else send forgot password email.
 
     """
+    campaign = request.args.get('c')
```

Comment:
For readability (and to avoid collision with other options added in the future), consider changing the name of this URL argument from `c` to `campaign`.

ref: http://cosdev.readthedocs.org/en/latest/style_guides/python.html#general-naming-guidelines